src/pages/DonatePage.js

```diff
@@ -0,0 +1,140 @@
+import { useContext, useState } from "react";
+
+import Button from "@material-ui/core/Button";
+import Container from "@material-ui/core/Container";
+import Paper from "@material-ui/core/Paper";
+import Typography from "@material-ui/core/Typography";
+import WhatshotIcon from "@material-ui/icons/Whatshot";
+
+import InternalLink from "../components/InternalLink";
+import User from "../components/User";
+import firebase, { customerPortal } from "../firebase";
+import useFirebaseRef from "../hooks/useFirebaseRef";
+import { UserContext } from "../context";
+import { patronCheckout } from "../stripe";
+
+function DonatePage() {
+  const user = useContext(UserContext);
+  const email = firebase.auth().currentUser.email;
```

Comment:
When testing, I found that if I'm on /donate page logged in with some email, then try to log out, an error page with "TypeError: Cannot read property 'email' of null" comes up. Perhaps we should get currentUser by itself and check if it's null first?

Reply:
Good catch!